hazelcast/src/test/java/com/hazelcast/client/protocol/ClientExceptionFactoryTestSimple.java

```diff
@@ -0,0 +1,78 @@
+package com.hazelcast.client.protocol;
+
+import com.hazelcast.client.impl.protocol.ClientExceptionFactory;
+import com.hazelcast.client.impl.protocol.ClientExceptionFactory.ExceptionFactory;
+import com.hazelcast.client.impl.protocol.ClientProtocolErrorCodes;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.test.HazelcastTestSupport.ignore;
+import static org.junit.Assert.fail;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+// It's called Simple, because the ClientExceptionFactoryTest is parametrized
+public class ClientExceptionFactoryTestSimple {
+
+    private ClientExceptionFactory exceptionFactory = new ClientExceptionFactory(false);
+
+    private static class MyException extends Exception {}
+
+    @Test
+    public void testDuplicateErrorCode() {
+        try {
+            exceptionFactory.register(ClientProtocolErrorCodes.ARRAY_INDEX_OUT_OF_BOUNDS,
```

Comment:
it would be better to use `ExpectedException` `Rule` here and in the other tests, rather than try catch blocks